AVInput/AVInput.cpp

```diff
@@ -224,5 +327,923 @@ string AVInput::currentVideoMode(bool &success)
+    {
+        device::HdmiInput::getInstance().getSupportedGameFeatures (supportedFeatures);
+        for (size_t i = 0; i < supportedFeatures.size(); i++)
+        {
+            LOGINFO("Supported Game Feature [%zu]:  %s\n",i,supportedFeatures.at(i).c_str());
+        }
+    }
+    catch (const device::Exception& err)
+    {
+        LOG_DEVICE_EXCEPTION0();
+    }
+
+    if (supportedFeatures.empty()) {
+        return Core::ERROR_GENERAL;
+    }
+    else {
+        setResponseArray(response, "suppoortedGameFeatures", supportedFeatures);
+        return Core::ERROR_NONE;
+    }
+}
+
+uint32_t AVInput::getGameFeatureStatusWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    string sPortId = parameters["portId"].String();
+    string sGameFeature = parameters["gameFeature"].String();
+    int portId = 0;
+
+    LOGINFOMETHOD();
+    if (parameters.HasLabel("portId") && parameters.HasLabel("gameFeature"))
+    {
+        try {
+            portId = stoi(sPortId);
+        }catch (const device::Exception& err) {
+            LOG_DEVICE_EXCEPTION1(sPortId);
+            return Core::ERROR_BAD_REQUEST;
+        }
+    }
+    else {
+        LOGWARN("Required parameters are not passed");
+        return Core::ERROR_BAD_REQUEST;
+    }
+
+    if (strcmp (sGameFeature.c_str(), "ALLM") == 0)
+    {
+        bool allm = getALLMStatus(portId);
+        LOGWARN("AVInput::getGameFeatureStatusWrapper ALLM MODE:%d", allm);
+        response["mode"] = allm;
+    }
+    else
+    {
+        LOGWARN("AVInput::getGameFeatureStatusWrapper Mode is not supported. Supported mode: ALLM");
+        return Core::ERROR_BAD_REQUEST;
+    }
+    return Core::ERROR_NONE;
+}
+
+bool AVInput::getALLMStatus(int iPort)
+{
+    bool allm = false;
+
+    try
+    {
+        device::HdmiInput::getInstance().getHdmiALLMStatus (iPort, &allm);
+        LOGWARN("AVInput::getALLMStatus ALLM MODE: %d", allm);
+    }
+    catch (const device::Exception& err)
+    {
+        LOG_DEVICE_EXCEPTION1(std::to_string(iPort));
+    }
+    return allm;
+}
+
+uint32_t AVInput::getRawSPDWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    LOGINFOMETHOD();
+
+    string sPortId = parameters["portId"].String();
+    int portId = 0;
+    if (parameters.HasLabel("portId"))
+    {
+        try {
+            portId = stoi(sPortId);
+        }catch (const device::Exception& err) {
+            LOG_DEVICE_EXCEPTION1(sPortId);
+            return Core::ERROR_BAD_REQUEST;
+        }
+    }
+    else {
+        LOGWARN("Required parameters are not passed");
+        return Core::ERROR_BAD_REQUEST;
+    }
+
+    string spdInfo = getRawSPD (portId);
+    response["HDMISPD"] = spdInfo;
+    if (spdInfo.empty()) {
+        return Core::ERROR_GENERAL;
+    }
+    else {
+        return Core::ERROR_NONE;
+    }
+}
+
+uint32_t AVInput::getSPDWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    LOGINFOMETHOD();
+
+    string sPortId = parameters["portId"].String();
```

Comment:
please use `parameters["portId"].Number()` if it is number

Reply:
Modified code to use Number()